glean-core/python/glean/glean.py

```diff
@@ -132,56 +132,60 @@ def initialize(
 
         cls._upload_enabled = upload_enabled
 
-        cfg = _ffi.make_config(
-            cls._data_dir,
-            application_id,
-            cls._upload_enabled,
-            configuration.max_events,
-        )
+        # Use `Glean._execute_task` rather than `Glean.launch` here, since we
+        # never want to put this work on the `Dispatcher._preinit_queue`.
+        @Dispatcher._execute_task
```

Comment:
Please note that everything before this point should be thread-safe (for consistency with other platforms), but I don't see this being enforced in any way. Should we be worried? If not, can we document why we believe we shouldn't (e.g. some GIL detail I do not know about).

Reply:
I filed this bug to handle this as a follow-up: https://bugzilla.mozilla.org/show_bug.cgi?id=1641564